component/service/src/main/java/org/exoplatform/social/rest/api/ProfileSettingsRest.java

```diff
@@ -0,0 +1,198 @@
+    @ApiResponse(responseCode = "401", description = "Unauthorized operation") })
+  public Response getPropertySettings(@Context
+  UriInfo uriInfo, @Context
+  Request request) {
+    long currentIdentityId = RestUtils.getCurrentUserIdentityId();
+    if (currentIdentityId == 0) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    try {
+      return Response.ok(EntityBuilder.buildEntityProfilePropertySettingList(profilePropertySettingsService.getPropertySettings(), labelService, profilePropertySettingsService, ProfilePropertySettingsService.LABELS_OBJECT_TYPE)).build();
+    }catch (Exception e) {
+      LOG.error("An error occurred while getting list of settings", e);
+      return Response.status(HTTPStatus.INTERNAL_ERROR).build();
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @POST
+  @RolesAllowed("administrators")
+  @Operation(
+          summary = "Creates a Profile property setting",
+          method = "POST",
+          description = "\"Creates a Profile property setting.")
+  @ApiResponses(value = {
+          @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+          @ApiResponse(responseCode = "500", description = "Internal server error"),
+          @ApiResponse(responseCode = "400", description = "Invalid query input"),
+          @ApiResponse(responseCode = "409", description = "Conflict"),
+          @ApiResponse(responseCode = "401", description = "Unauthorized operation")})
+  public Response createPropertySetting(@Context UriInfo uriInfo,
+                                        @RequestBody(description = "Profile property setting object to be created"
+                                                , required = true) ProfilePropertySettingEntity profilePropertySettingEntity) {
+    long currentIdentityId = RestUtils.getCurrentUserIdentityId();
+    if (currentIdentityId == 0) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    if (profilePropertySettingEntity == null || profilePropertySettingEntity.getPropertyName() == null) {
+      return Response.status(Status.BAD_REQUEST).entity("Profile property setting is null or property name not provided").build();
+    }
+    try {
+      ProfilePropertySetting newProfilePropertySetting = profilePropertySettingsService.createPropertySetting(EntityBuilder.buildProfilePropertySettingFromEntity(profilePropertySettingEntity));
+      if (!profilePropertySettingEntity.getLabels().isEmpty()) {
+        labelService.createLabels(profilePropertySettingEntity.getLabels(), ProfilePropertySettingsService.LABELS_OBJECT_TYPE, String.valueOf(newProfilePropertySetting.getId()));
+      }
+      return Response.ok().build();
+    } catch (ObjectAlreadyExistsException ex) {
+      LOG.warn("property.already.exist", ex);
+      return Response.status(Status.CONFLICT).build();
+    }catch (Exception ex) {
+      LOG.warn("Failed to create a new Property setting", ex);
```

Comment:
this type of unxpected exceptions has to be loggued with the provided information by user to help on diagnosis and troubleshooting on production environments especially. (to apply everywhere in REST Services)

Reply:
removed